### PR TITLE
Separate AseDbType from DbType

### PR DESCRIPTION
**Pull request by senseibaka**
The `AseDbType` started off being an enum whose values were informed by (pinned to) those of `DbType`, which lead to some enum values being the same/indistinguishable (like `Image` vs `Binary`).

This generally isn't a big deal, but it makes it harder for callers to specify an exact `AseDbType` and see the behaviour they're expecting (generally in edge cases, admittedly).

So, this change seeks to right that wrong, and at the same time address issues #59 and #62 .

* Introduced tests which assert that our `AseDbType`'s numeric values are equivalent to the reference driver's
* Introduced tests related to issues #59 and #62
* Fixed issues related to introduced tests
* Had to implement the `TDS_BIGDATETIMEN` type for parameter transmission, which involved inspecting Ribo output and figuring out that in this format, the date/time is encoded as an 8 byte number representing the number of microseconds since year 0 (1 BC?). Turns out that the reference driver prefers to use big datetime even when transmitting times. Probably because it's a better resolution?

### CODE REVIEW

**Comment by senseibaka** [comment]
Note: don't merge just yet, I might be adding a few more tests/fixes

**Comment by senseibaka** [comment]
So, this PR now fixes issues:
* #51 - Unicode/string issues (\u2000, \u2001, and weird behaviour when charset isn't specified in the connection string)
* #59 - Date / Time related issues
* #62 - Image vs Binary type handling of null/empty

**Comment by senseibaka** [comment]
Yep, I reckon it's ready for review / merge

File: src/AdoNetCore.AseClient/Internal/TypeMap.cs [hunk 1]
```diff
@@ -340,5 +342,96 @@ public static AseDbType GetAseDbType(FormatItem format)
...
+            {AseDbType.Time, DbType.Time},
+            {AseDbType.TimeStamp, DbType.Binary},
+            {AseDbType.TinyInt, DbType.Byte},
+            {AseDbType.UniChar, DbType.StringFixedLength},
+            {AseDbType.Unitext, DbType.String},
+            {AseDbType.UniVarChar, DbType.String},
+            {AseDbType.UnsignedBigInt, DbType.UInt64},
+            {AseDbType.UnsignedInt, DbType.UInt32},
+            {AseDbType.UnsignedSmallInt, DbType.UInt16},
+            {AseDbType.Unsupported, DbType.Guid},
```
Comment (sheikhjabootie) [inline src/AdoNetCore.AseClient/Internal/TypeMap.cs]: Why are we mapping Unsupported to Guid?

File: src/AdoNetCore.AseClient/Internal/TypeMap.cs [hunk 2]
```diff
@@ -340,5 +342,96 @@ public static AseDbType GetAseDbType(FormatItem format)
...
+        };
+
+        public static DbType GetDbType(AseDbType aseDbType)
+        {
+            return AseDbTypeToDbTypeMap.ContainsKey(aseDbType)
+                ? AseDbTypeToDbTypeMap[aseDbType]
+                : default(DbType);
+        }
+
+        public static AseDbType CleanupAseDbType(AseDbType aseDbType)
```
Comment (sheikhjabootie) [inline src/AdoNetCore.AseClient/Internal/TypeMap.cs]: @senseibaka - this looks a little hacky/obscure. Maybe a code comment as to your rationale here?

Comment (senseibaka) [reply src/AdoNetCore.AseClient/Internal/TypeMap.cs]: Good point, will do.

-- FYI it's because the reference driver doesn't distinguish between DateTime/BigDateTime (and they're both transmitted as `TDS_BIGDATETIMEN` type, if that makes sense).

Comment (sheikhjabootie) [reply src/AdoNetCore.AseClient/Internal/TypeMap.cs]: Got it. Figured you'd have a reason. Stick that in the code so the next guy isn't wondering ;-)

File: src/AdoNetCore.AseClient/Internal/TypeMap.cs (hunk 1 above)
Comment (senseibaka) [reply src/AdoNetCore.AseClient/Internal/TypeMap.cs]: For consistency with the reference driver -- It appears to treat `Guid` as a synonym for unsupported.

If the user explicitly sets the DbType to Guid, then they can still use the Guid type (as demonstrated by `SimpleQueryTests.SelectGuid_Parameter_ShouldWork`)
